src/WithPagination.php

```diff
@@ -75,8 +86,10 @@ public function resetPage($pageName = 'page')
 
     public function setPage($page, $pageName = 'page')
     {
-        $page = intval($page);
-        $page = $page <= 0 ? 1 : $page ;
+        if (is_numeric($page)){
```

Comment:
since cursor pagination use string as the identifier we should not cast the $page to intger all the time